synapse/util/check_dependencies.py

```diff
@@ -66,6 +66,17 @@ def _is_dev_dependency(req: Requirement) -> bool:
     )
 
 
+def _should_ignore_runtime_requirement(req: Requirement) -> bool:
+    # This is a build-time dependency. Irritatingly, `poetry build` ignores the
+    # requirements listed in the [build-system] section of pyproject.toml, so in order
+    # to support `poetry install --no-dev` we have to mark it as a runtime dependency.
+    # Workaround this by ignoring it here. (It might be slightly cleaner to put
+    # `setuptools_rust` in a `build` extra or similar, but . But for now I'
```

Comment:
some words have gotten lost here

Reply:
:facepalm: Guh, I think this is a sign I need to eat. Thanks, will amend